code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -459,11 +462,16 @@
 		else if(isliving(M) && !faction_check_mob(M)) //shyness check. we're not shy in front of things that share a faction with us.
 			return //we never mate when not alone, so just abort early
 
-	if(alone && partner && children < 3)
+	// The children check here isn't the total number of offspring, but the
+	// number of offspring within a visible 7-tile radius. It doesn't seem very
+	// effective at preventing population explosions but there you go.
+	if(partner && children < 3)
 		var/childspawn = pickweight(childtype)
 		var/turf/target = get_turf(loc)
 		if(target)
-			return new childspawn(target)
+			new childspawn(target)
+			current_offspring += 1
+			return
```

Comment:
You should return `new childspawn(target)` as there are functions calls that require it.

Reply:
Good catch, I'm just going to use the old return. I was trying to not increase the count until we were sure the spawn occurred but I don't think it matters.